content/blog/esp32-memory-map-101/index.md

```diff
@@ -0,0 +1,351 @@
+---
+title: "ESP32's family Memory Map 101"
+date: 2024-08-20T18:56:49+02:00
+tags: ["ESP32", "ESP32-S2", "ESP32-S3", "ESP32-C3", "ESP32-C6"]
+showAuthor: false
+authors:
+  - "marek-matej"
+---
+
+
+The performance and memory capabilities of the microcontrollers are increasing, and so is their complexity, especially when the users are responsible for mapping external memory chips such as Flash or SPIRAM by configuring a memory management unit.
+
+When I started to use Espressif ESP32’s series SoC’s with the [Zephyr RTOS](https://docs.zephyrproject.org/latest/index.html), quickly I realized that these SoC’s are a very different world, compared to my previous MCU experiences, which were mostly ARM-related. ARM Cortex-M’s - in most cases - uses Von-Neumann memory architecture, while the ESP32’s series of microcontrollers uses the Harvard memory architecture, which adds complexity.
+
+The primary source of knowledge about any SoC was always the datasheet or Technical Reference Manual of the part. Indeed it still is, but I found the ESP32's TRM — although very useful — quite challenging to read. Especially because there is a significant portion of important information buried in the long textual descriptions.
```

Comment:
Very true! it is also quite challenging to structure or edit this documentation :)

Reply:
...and the main motivation for this article as well :)